### PR TITLE
Adding support for username/password env. vars.

### PR DESCRIPTION
**Pull request by apahim**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,8 +10,6 @@ COPY nginx.conf auth.htpasswd docker-entrypoint.sh ./
 
 USER root
 RUN chmod 777 /docker-entrypoint.sh
-RUN chmod -R 777 /usr/share/nginx/html/
-RUN echo "nginx on CentOS7" > /usr/share/nginx/html/index.html
```
Comment (jmelis) [inline Dockerfile:14]: Why are you removing this? Not against removing it, but want to understand why.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```
+$ htpasswd -bn foo bar
+foo:$apr1$00xPWwOg$sQRfZdrPbGiYz/qUPA8hX1
+$ docker run --rm -it -e HTPASSWD='foo:$apr1$00xPWwOg$sQRfZdrPbGiYz/qUPA8hX1' -e FORWARD_HOST=example.com -p 8080:8080 ng
+```
+
+ - Give the container the username and password:
+
+```
+$ docker run --rm -it -e BASIC_AUTH_USERNAME='foo' -e BASIC_AUTH_PASSWORD='bar' -e FORWARD_HOST=example.com -p 8080:8080 ng
```
Comment (jmelis) [inline README.md:27]: I'm not sure I like the idea of having the password in plain text in an env var... if an attacker could gain access to dumping the env, or to access an env var, they could exploit this.

What's the use case for this vs simply using `HTPASSWD`?

File: Dockerfile (hunk 1 above)
Comment (apahim) [reply Dockerfile:14]: it's broken (it's a separate commit)

File: README.md (hunk 2 above)
Comment (apahim) [reply README.md:27]: How is that different when compared to all the secrets we expose via Pods environment variables in our clusters?
What I want to achieve here is to have both the plain text password (for client access) and the ecripted on (for nginx configuration) in vault.

**Pull request merged**